probe/main.go

```diff
@@ -83,24 +83,22 @@ func main() {
 	)
 
 	// TODO provide an alternate implementation for Darwin.
```

Comment:
Also, the if processCache != nil line below can go too (I can't seem to comment on that line, sorry)